lib/beaker/host.rb

```diff
@@ -224,13 +225,29 @@ def is_x86_64?
     end
 
     def connection
-      @connection ||= SshConnection.connect( reachable_name,
+      # create new connection object if necessary
+      @connection ||= SshConnection.connect( { :ip => self['ip'], :vmhostname => self['vmhostname'], :hostname => @name },
                                              self['user'],
                                              self['ssh'], { :logger => @logger } )
+      # update connection information
+      if self['ip'] && (@connection.ip != self['ip'])
+        @connection.ip = self['ip']
+      end
+      if self['vmhostname'] && (@connection.vmhostname != self['vmhostname'])
+        @connection.vmhostname = self['vmhostname']
+      end
+      if @name && (@connection.hostname != @name)
+        @connection.hostname = @name
+      end
+      @connection
     end
 
     def close
       @connection.close if @connection
+      # update connection information
```

Comment:
The first ensures that the connection is closed, the second throws away the old object.